### PR TITLE
Improve building native extensions

### PR DESCRIPTION
**Pull request by jwerle**

### CODE REVIEW

File: src/cli/cli.cc
Comment (chicoxyzzy) [inline src/cli/cli.cc:1]: that was pretty hard to review

**Review by chicoxyzzy — COMMENTED** [review]
Linux build is failing because `resolvePromise` call  is not updated in the `linux.cc`

File: src/cli/cli.cc [hunk 1]
```diff
@@ -1712,7 +1712,7 @@ int main (const int argc, const char* argv[]) {
         }
 
         // only validate if config exists and we didn't exit early above
-        if (configExists) {
+        if (configExists && !equal(subcommand, "init") && !equal(subcommand, "env")) {
```
[the remark this answers is not in the document]
Comment (jwerle) [reply src/cli/cli.cc:1715]: it doesn't make sense to check for config for these commands as they do not depend on `socket.ini` being there

File: src/cli/cli.cc [hunk 2]
```diff
@@ -2000,10 +2000,25 @@ int main (const int argc, const char* argv[]) {
   });
 
  Options printBuildDirOptions = {
-    { { "--platform" }, true },
-    { { "--prod" }, true }
+    { { "--platform" }, true }, { { "--root" }, true}
```
[the remark this answers is not in the document]
Comment (chicoxyzzy) [reply src/cli/cli.cc:2003]: I think this is the last change pending before merge

**Review by chicoxyzzy — APPROVED** [review]
LGTM

**Pull request merged**